### PR TITLE
Let the renaming begin

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by cpcloud** [comment]
@mrocklin this is ready for review when you get a chance

File: README.rst [hunk 1]
```diff
@@ -74,11 +74,11 @@ method.  Similar functions exist for ``append``, to add to existing data, and
 * ``convert``: Transform dataset into new container
 * ``append``: Add dataset onto existing container
 * ``resource``: Given a URI find the appropriate data resource
-* ``into``: Call one of the above based on inputs.
+* ``odo``: Call one of the above based on inputs.
   E.g. ``into(list, (1, 2, 3)) -> convert(list, (1, 2, 3))``
```
Comment (mrocklin) [inline README.rst]: Do we want to swap these too in this pass?

File: docs/Makefile [hunk 2]
```diff
@@ -77,17 +77,17 @@ qthelp:
...
 
 devhelp:
 	$(SPHINXBUILD) -b devhelp $(ALLSPHINXOPTS) $(BUILDDIR)/devhelp
 	@echo
 	@echo "Build finished."
 	@echo "To view the help file:"
-	@echo "# mkdir -p $$HOME/.local/share/devhelp/into"
-	@echo "# ln -s $(BUILDDIR)/devhelp $$HOME/.local/share/devhelp/into"
+	@echo "# mkdir -p $$HOME/.local/share/devhelp/odo"
+	@echo "# ln -s $(BUILDDIR)/devhelp $$HOME/.local/share/devhelp/odo"
```
Comment (mrocklin) [inline docs/Makefile]: Thank heavens for command line find-replace

File: README.rst (hunk 1 above)
Comment (cpcloud) [inline README.rst]: Yeah, I may have missed a few places. Feel free to point them out.

File: docs/source/aws.rst [hunk 3]
```diff
@@ -58,22 +58,25 @@ To access an individual table simply append ``::`` plus the table name
...
 such, they are able to take advantage of conversions to types that don't have
 an explicit path defined for them. This allows us to do things like convert an
 S3 CSV to a pandas DataFrame
 
     .. code-block:: python
 
-       >>> df = into(pandas.DataFrame, 's3://mybucket/myfile.csv')
+       >>> import pandas as pd
+       >>> from odo import odo
+       >>> df = odo(pd.DataFrame, 's3://mybucket/myfile.csv')
```
Comment (mrocklin) [inline docs/source/aws.rst]: Swap

File: docs/Makefile (hunk 2 above)
Comment (cpcloud) [inline docs/Makefile]: `sed` to the rescue on this PR

File: docs/source/conf.py [hunk 4]
```diff
@@ -40,17 +40,17 @@
...
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 #
 # The short X.Y version.
-version = '0.1'
+version = '0.2.2'
 # The full version, including alpha/beta/rc tags.
-release = '0.1'
+release = '0.2.2'
```
Comment (mrocklin) [inline docs/source/conf.py]: from into import __version__  as version

File: docs/source/conf.py [hunk 5]
```diff
@@ -245,10 +245,10 @@
 # -- Options for Epub output ---------------------------------------------------
 
 # Bibliographic Dublin Core info.
-epub_title = u'into'
+epub_title = u'odo'
 epub_author = u'Matthew Rocklin'
```
Comment (mrocklin) [inline docs/source/conf.py]: We should probably remove the author line or switch it to something generic.

File: docs/source/datashape.rst [hunk 6]
```diff
@@ -90,18 +90,18 @@ And any composition of the above::
...
 ``dshape=`` keyword
 
 .. code-block:: python
 
-    >>> into(pd.DataFrame, 'accounts.csv',
+    >>> odo('accounts.csv', pd.DataFrame,
     ...      dshape='var * {name: string, balance: float64}')
 
-This removes all of the guessword from the ``into`` heuristics and.  This can
+This removes all of the guesswork from the ``odo`` heuristics.  This can
```
Comment (mrocklin) [inline docs/source/datashape.rst]: Thank you for also doing the general sanity pass on things.

File: docs/source/hdfs.rst [hunk 7]
```diff
@@ -19,12 +19,13 @@ Simple and complex examples follow::
...
 Alternatively you may want to pass authentication information through keyword
-arguments to the into function as in the following example
+arguments to the ``odo`` function as in the following example
 
 .. code-block:: python
 
-   >>> into('hfds://hostname:myfile.csv', 'localfile.csv',
-   ...      port=14000, user='hdfs')
+   >>> from odo import odo
+   >>> odo('localfile.csv', 'lhfds://hostname:myfile.csv',
```
Comment (mrocklin) [inline docs/source/hdfs.rst]: lhfds

File: docs/source/index.rst [hunk 8]
```diff
@@ -1,19 +1,19 @@
+Odo: Shapeshifting for your data
+================================
 
-Into
-====
-
-``into`` takes two arguments, a target and a source for a data transfer.
+``odo.into`` takes two arguments, a target and a source for a data transfer.
```
Comment (mrocklin) [inline docs/source/index.rst]: odo.odo

Comment (mrocklin) [inline docs/source/index.rst]: or just odo

File: setup.py [hunk 9]
```diff
@@ -14,22 +15,22 @@ def find_data_files(where, exts):
 
 exts = ('*.h5', '*.csv', '*.xls', '*.xlsx', '*.db', '*.json', '*.gz', '*.hdf5',
         '*.sas7bdat')
-package_data = [x.replace('into' + os.sep, '') for x in
-                    find_data_files('into', exts)]
+package_data = [x.replace('odo' + os.sep, '') for x in
+                find_data_files('into', exts)]
```
Comment (mrocklin) [inline setup.py]: odo

File: setup.py [hunk 10]
```diff
@@ -14,22 +15,22 @@ def find_data_files(where, exts):
...
       author='Matthew Rocklin',
       author_email='mrocklin@continuum.io',
       license='BSD',
-      keywords='into data conversion hdf5 sql blaze',
+      keywords='odo data conversion hdf5 sql blaze',
       packages=find_packages(),
       install_requires=list(open('requirements.txt').read().strip().split('\n')),
       long_description=(open('README.rst').read() if os.path.exists('README.rst')
                         else ''),
       package_data={'into': package_data},
```
Comment (mrocklin) [inline setup.py]: odo?

**Comment by mrocklin** [comment]
As I was going through this I kept thinking "Gosh I'm glad I didn't have to do this" :)

**Comment by cpcloud** [comment]
fixed typos and addressed your comments

**Comment by mrocklin** [comment]
OK, I'm going to have to stall on this until a bit later tonight when I'll go through the code itself.  Excited about this though.  What are some of the other things we'll need to update outside of the repo?

**Comment by cpcloud** [comment]
We'll need to update blaze and Dask, change the actual repo name, and send out a mailing list message telling people to use odo instead of into for new code.

**Comment by cpcloud** [comment]
Happy to do those things

**Comment by cpcloud** [comment]
And of course cutting a release is probably in order as well

**Comment by cpcloud** [comment]
@mrocklin made a new 0.3.0 milestone starting with this PR. seems like a more significant change than a point release

**Comment by mrocklin** [comment]
Generally +1 on this.  Presumably there will be some friction with various other systems.  You seem pretty on top of this though.  If I were you I would probably merge and screw around on master for a while.  It might be worth checking out the binstar-build setup as a good test case for how well we interact with the build ecosystem.  We should also register `odo` on PyPI.

**Comment by cpcloud** [comment]
things left to do/to be done after merge

- [ ] update blaze
- [ ] update dask
- [ ] register on PyPI `python setup.py register` will do this
- [ ] checkout binstar-build
- [ ] futz around a bit on master

**Comment by mrocklin** [comment]
If anything on that list is foreign to you I'm happy to handle it to speed things up.

**Comment by cpcloud** [comment]
maybe update dask and register on PyPI? everything else i can do

**Comment by mrocklin** [comment]
Will do.  Let me know when a version is tagged (and no rush).

**Pull request merged**